LibreNMS/Alert/Transport/Msteams.php

```diff
@@ -42,9 +41,11 @@ public function contactMsteams($obj, $opts)
             'Expect:'
         ));
         curl_setopt($curl, CURLOPT_POSTFIELDS, json_encode($data));
+        if ($this->config['use-json'] === 'on' && $obj['name'] !== 'Test-Rule') {
```

Comment:
What's the reference to Test-Rule here? We shouldn't base conditions on what people might call alert rules.

Reply:
It comes from the array in /opt/librenms/includes/html/forms/test-transport.inc.php for the test message.

```
...
$obj = array(
    "hostname"  => $tmp[0]['hostname'],
    "device_id" => $tmp[0]['device_id'],
    "sysDescr" => $tmp[0]['sysDescr'],
    "version" => $tmp[0]['version'],
    "hardware" => $tmp[0]['hardware'],
    "location" => $tmp[0]['location'],
    "title" => "Testing transport from " . Config::get('project_name'),
    "elapsed"   => "11s",
    "id"        => "000",
    "faults"    => false,
    "uid"       => "000",
    "severity"  => "critical",
    "rule"      => "macros.device = 1",
    "name"_      => "Test-Rule",
    "string"      => "#1: test => string;",
    "timestamp" => date("Y-m-d H:i:s"),
    "contacts"  => $tmp['contacts'],
    "state"     => "1",
    "msg"       => "This is a test alert",
...
```

Maybe it should use the id instead? Since it is probably likely that somebody would name their rule "Test-Rule".

Reply:
I changed it to use $obj['id'] !== '000' since there should never be an alert with that ID except the test transport alert.